packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { describe, it, vi } from 'vitest';
+import * as ApiKubernetesModule from '@/api/data/kubernetes';
+import {
+  useAllKube,
+  useKubernetesCluster,
+  useKubes,
+  useRenameKubernetesCluster,
+  useResetKubeConfig,
+  useUpdateKubePolicy,
+} from './useKubernetes';
+import { wrapper } from '@/wrapperRenders';
+import { TKube } from '@/types';
+
+vi.mock('@/api/data/kubernetes', async () => {
+  const mod = await vi.importActual('@/api/data/kubernetes');
+  return {
+    ...mod,
+    getAllKube: vi.fn(),
+    getKubernetesCluster: vi.fn(),
+    updateKubernetesCluster: vi.fn(),
+    resetKubeConfig: vi.fn(),
+    updateKubePolicy: vi.fn(),
+    getClusterRestrictions: vi.fn(),
+    getOidcProvider: vi.fn(),
+    postKubeConfig: vi.fn(),
+    updateKubeVersion: vi.fn(),
+    terminateCluster: vi.fn(),
+    resetCluster: vi.fn(),
+    addOidcProvider: vi.fn(),
+    updateOidcProvider: vi.fn(),
+    removeOidcProvider: vi.fn(),
+    getSubscribedLogs: vi.fn(),
+    createSubscription: vi.fn(),
+    deleteSubscription: vi.fn(),
+  };
+});
```

Comment:
You dont need the mock of the module here, since spyOn the needed funtions bellow